metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py

```diff
@@ -621,14 +622,37 @@ def generate_partition_profiler_query(
 
         partition = self.get_latest_partition(schema, table)
         if partition:
-            partition_ts: Union[datetime.datetime, datetime.date]
+            partition_where_clause: str
             if not partition_datetime:
                 partition_datetime = parser.parse(partition.partition_id)
             logger.debug(f"{table} is partitioned and partition column is {partition}")
             if partition.data_type in ("TIMESTAMP", "DATETIME"):
-                partition_ts = partition_datetime
+                duration: relativedelta
+                # if yearly partitioned,
+                if len(partition.partition_id) == 4:
```

Comment:
Please add unit tests for this. Maybe extracting the code inside the `if` block into a separate function? That will give examples to why we have this `len` checks in here and validate that this works. Currently after this change it is hard to understand what we are doing here.

Reply:
That makes sense. I will create unit tests for this.